docs/versioned_docs/version-2.11.0/concepts/cstor.md

```diff
@@ -0,0 +1,324 @@
+
+**Delayed snapshots**
+
+In cStor, snapshots are taken only when the volume replicas are in quorum. For example, as soon as the volume is provisioned on cStor, the volume will be in ready state but the quorum attainment may take few minutes. Snapshot commands during this period will be delayed or queued till the volumes replicas attain quorum. The snapshot commands received by the target are also delayed when the cStor volume is marked read-only because of no-quorum.
+
+## Troubleshooting areas
+
+Following are most commonly observed areas of troubleshooting
+
+1. **iSCSI tools are not installed or iscsi.d service is not running**
+
+   **Symptom:**
+
+   If iSCSI tools are not installed on the nodes, after launching the application with PVC pointing to OpenEBS provisioner,  you may observe the application is in "ContainerCreating" state. 
+
+   `kubectl describe pod <app pod> ` may give an error that looks like the following
+
+   ```
+   MountVolume.WaitForAttach failed for volume "pvc-33408bf6-2307-11e9-98c4-000d3a028f9a" : executable file not found in $PATH
+   ```
+
+   **Resolution**: 
+
+   Install iSCSI tools and make sure that iSCSI service is running. See [iSCSI installation](/docs/next/prerequisites.html)
+
+2. **Multi-attach error is seen in the logs**
+
+   **Symptom:**
+
+   In a node failure case, it is sometimes observed that the application pods that were running on the node that went down are moved, the cStor target pods are also moved to other nodes, but the application throws up an error similar to below:
+
+   ```
+   Warning  FailedAttachVolume  4m15s  attachdetach-controller            
+   Multi-Attach error for volume "pvc-d6d52184-1f24-11e9-bf1d-0a0c03d0e1ae" 
+   Volume is already exclusively attached to one node and can't be attached to another
+   ```
+
+   **Resolution :**
+
+   This is because of the Kubernetes limitation [explained above](#known-limitations). OpenEBS CSI driver will resolve this issue. See [roadmap](#cstor-roadmap). In this situation if the node is stuck in a non-responsive state and if the node has to be permanently deleted, remove it from the cluster.  
+
+3. **Application is in ContainerCreating state, observed connection refused error**
+
+   **Symptom:**
+
+   Application is seen to be in the `ContainerCreating` state, with kubectl describe showing `Connection refused` error to the cStor PV. 
+
+   **Resolution**: 
+
+   This error eventually could get rectified on the further retries, volume gets mounted and application is started. This error is usually seen when cStor target takes some time to initialize  on low speed networks as it takes time to download cStor image binaries from repositories ( or )  or because the cstor target is waiting for the replicas to connect and establish quorum. If the error persists beyond 5 minutes, logs need to be verified, contact <a href="/docs/next/support.html" target="_blank">OpenEBS Community</a> for support.
```

Comment:
```suggestion
   This error eventually could get rectified on the further retries, volume gets mounted and application is started. This error is usually seen when cStor target takes some time to initialize  on low speed networks as it takes time to download cStor image binaries from repositories ( or )  or because the cstor target is waiting for the replicas to connect and establish quorum. If the error persists beyond 5 minutes, logs need to be verified, contact [OpenEBS Community](/docs/introduction/community) for support.
```